tests/multi_ptr/multi_ptr_implicit_conversions.h

```diff
@@ -0,0 +1,237 @@
+    // the minimal part of compilation path for better test support
+
+    using invoke_conversion_t = std::conditional_t<
+        has_implicit_conversion_available,
+        invoke_implicit_conversion<from_multi_ptr_t, dest_multi_ptr_t>,
+        avoid_implicit_conversion<from_multi_ptr_t>>;
+
+    {
+      sycl::range r(1);
+      sycl::buffer<bool> res_buf(&res, r);
+      sycl::buffer<T> expected_val_buffer(&value, r);
+      queue.submit([&](sycl::handler &cgh) {
+        auto res_acc =
+            res_buf.template get_access<sycl::access_mode::write>(cgh);
+        auto expected_val_acc =
+            expected_val_buffer.template get_access<sycl::access_mode::read>(
+                cgh);
+        cgh.single_task([=] {
+          from_multi_ptr_t mptr_from(expected_val_acc);
+
+          // from cppreference.com
+          //  Implicit conversions are performed whenever an
+          //  expression of some type T1 is used in context
+          //  that does not accept that type, but accepts some
+          //  other type T2; in particular:
+          //  <...>
+          //  - when initializing a new object of type T2, including return
+          //  statement in a function returning T2;
+          //  <...>
+          dest_multi_ptr_t mptr_dest = invoke_conversion_t{}(mptr_from);
+
+          // for cases, when dest_multi_ptr_t equals to multi_ptr<void>
+          T value_dest = static_cast<T>(*(mptr_dest.get()));
+
+          res_acc[0] = (value_dest == expected_val);
+        });
+      });
+    }
+    CHECK(res);
+  }
+
+ public:
+  void operator()(const std::string &type_name,
+                  const std::string &address_space_str,
+                  const std::string &is_decorated_str) {
+    SECTION(section_name("Verifying implicit conversion from "
+                         "multi_ptr<T,address_space,decorated>")
+                .with("T", type_name)
+                .with("address_space", address_space_str)
+                .with("decorated", is_decorated_str)
+                .create()) {
+      SECTION("Implicit conversion from multi_ptr<T> to multi_ptr<void>") {
+        using from_multi_ptr_t = sycl::multi_ptr<T, address_space, decorated>;
+
+        using dest_multi_ptr_not_decorated_t =
+            sycl::multi_ptr<void, address_space, sycl::access::decorated::no>;
+        using dest_multi_ptr_decorated_t =
+            sycl::multi_ptr<void, address_space, sycl::access::decorated::yes>;
+
+        SECTION("Conversion to multi_ptr<void, decorated::yes>") {
+          preform_implicit_conversion_test<from_multi_ptr_t,
+                                           dest_multi_ptr_decorated_t>();
+        }
+        SECTION("Conversion to multi_ptr<void, decorated::no>") {
+          preform_implicit_conversion_test<from_multi_ptr_t,
+                                           dest_multi_ptr_not_decorated_t>();
+        }
+      }
+      // clang-format off
```

Comment:
This comment is required?

Reply:
Looks like we don't need it. I will remove it.

Reply:
Done [here](https://github.com/KhronosGroup/SYCL-CTS/pull/362/commits/5c36543f8210375fd474006b210658bef4ed300e).